src/C++/helpers.h

```diff
@@ -62,3 +62,23 @@ void print_title(const string & title){
   cout << concol::BOLDMAGENTA << title << concol::RESET << endl;
   cout << endl;
 }
+
+//Copy of APS2::print_firmware_version
+string print_firmware_version(uint32_t version_reg) {
+	std::ostringstream ret;
+
+	uint32_t minor{version_reg & 0xff};
+	uint32_t major{(version_reg >> 8) & 0xf};
+	uint32_t note_nibble{(version_reg >> 12) & 0xf};
```

Comment:
Yep, I can't really disagree with you.  It allows the compiler to catch some unintended narrowing but that's usually more annoying than helpful.